include/gsl/span

```diff
@@ -18,7 +18,7 @@
 #define GSL_SPAN_H
 
 #include "./assert"   // for Expects
-#include "./byte"     // for byte
+#include "./byte"     // for BYTE_TYPE
```

Comment:
It worries me if that change is necessary. That would mean that users using GSL might also need to use `BYTE_TYPE` instead of `gsl::byte`, which in my opinion is a change for the worse.